src/mlj_flux.jl

```diff
@@ -130,29 +157,90 @@ function MLJFlux.shape(model::LaplaceRegression, X, y)
     return (n_input, n_output)
 end
 
+
+
+"""
+    MLJFlux.build(model::LaplaceClassification, rng, shape)
+
+Builds an MLJFlux model for Laplace classification compatible with the dimensions of the input and output layers specified by `shape`.
+
+# Arguments
+- `model::LaplaceClassification`: The Laplace classification model.
+- `rng`: A random number generator to ensure reproducibility.
+- `shape`: A tuple or array specifying the dimensions of the input and output layers.
+
+# Returns
+- The constructed MLJFlux model, compatible with the specified input and output dimensions.
+"""
 function MLJFlux.build(model::LaplaceClassification, rng, shape)
     #chain
     chain = Flux.Chain(MLJFlux.build(model.builder, rng, shape...), model.finaliser)
 
     return chain
 end
 
+
+
+"""
+    MLJFlux.build(model::LaplaceRegression, rng, shape)
+
+Builds an MLJFlux model for Laplace regression compatible with the dimensions of the input and output layers specified by `shape`.
+
+# Arguments
+- `model::LaplaceRegression`: The Laplace regression model.
+- `rng`: A random number generator to ensure reproducibility.
+- `shape`: A tuple or array specifying the dimensions of the input and output layers.
+
+# Returns
+- The constructed MLJFlux model, compatible with the specified input and output dimensions.
+"""
 function MLJFlux.build(model::LaplaceRegression, rng, shape)
-    #chain
+
     chain = MLJFlux.build(model.builder, rng, shape...)
 
     return chain
 end
 
+
+"""
+    MLJFlux.fitresult(model::LaplaceClassification, chain, y)
+
+Computes the fit result for a Laplace classification model, returning the model chain and the number of unique classes in the target data.
+
+# Arguments
+- `model::LaplaceClassification`: The Laplace classification model to be evaluated.
+- `chain`: The trained model chain.
+- `y`: The target data, typically a vector of class labels.
+
+# Returns
+- A tuple containing:
+  - The model chain.
+  - The number of unique classes in the target data `y`.
+"""
 function MLJFlux.fitresult(model::LaplaceClassification, chain, y)
     return (chain, length(unique(y)))
 end
 
+
+"""
+    MLJFlux.fitresult(model::LaplaceRegression, chain, y)
+
+Computes the fit result for a Laplace Regression model, returning the model chain and the number of output variables in the target data.
+
+# Arguments
+- `model::LaplaceRegression`: The Laplace Regression model to be evaluated.
+- `chain`: The trained model chain.
+- `y`: The target data, typically a vector of class labels.
+
+# Returns
+- A tuple containing:
+  - The model chain.
+  - The number of unique classes in the target data `y`.
+"""
 function MLJFlux.fitresult(model::LaplaceRegression, chain, y)
     return (chain, size(y))
 end
 
-#########################################  fit and predict for classification
 
 """
     MLJFlux.fit!(model::LaplaceClassification, chain,penalty,optimiser,epochs, verbosity, X, y)
```

Comment:
I've had another look at this and it looks like you're not properly overloading the `MLJFlux.fit!` function. You need to use the same function signature as in MLJFlux, otherwise the default `MLJFlux.fit!` method is called. This also seems to be what leads to the error when your custom `predict` method is called (the fitresult containing the `la::Laplace` object is never created, instead you're getting a conventional neural network).

Reply:
[From phone]  but if I use the same signature and pass epochs, optimiser, batch etc as arguments, what's the reason for saving them in the laplaceregression struct?

Reply:
Also mljflux explicitly ask for those parameters to be added in the struct

Reply:
Because `MLJFlux` passes these arguments on under the hood: https://github.com/FluxML/MLJFlux.jl/blob/07e01b7f41bc3dd870dd08c7c97114cc45f91f5e/src/mlj_model_interface.jl#L124

Reply:
So how does mljbase.fit know  that it has to  call mljflux.fit if I do not overload it?

Reply:
In the same file I just referenced, the MLJBase.fit method is overloaded: https://github.com/FluxML/MLJFlux.jl/blob/07e01b7f41bc3dd870dd08c7c97114cc45f91f5e/src/mlj_model_interface.jl#L93

It is overloaded for all MLJFlux models and the our models are subtypes of MLJFlux models. That is exactly why we don't need to overload any MLJBase functionality.